kimchi/src/verifier.rs

```diff
@@ -762,28 +763,39 @@ pub fn verify<G, EFqSponge, EFrSponge>(
     group_map: &G::Map,
     verifier_index: &VerifierIndex<G>,
     proof: &ProverProof<G>,
+    public_input: &Vec<G::ScalarField>,
 ) -> Result<()>
 where
     G: KimchiCurve,
     G::BaseField: PrimeField,
     EFqSponge: Clone + FqSponge<G::BaseField, G, G::ScalarField>,
     EFrSponge: FrSponge<G::ScalarField>,
 {
-    let proofs = vec![(verifier_index, proof)];
+    let proofs = vec![BatchedProof {
+        verifier_index,
+        proof,
+        public_input,
+    }];
     batch_verify::<G, EFqSponge, EFrSponge>(group_map, &proofs)
 }
 
+/// Struct equivalent of tuple (verifier index, Plonk proof, public input)
+pub struct BatchedProof<'a, G: KimchiCurve> {
+    pub verifier_index: &'a VerifierIndex<G>,
+    pub proof: &'a ProverProof<G>,
+    pub public_input: &'a Vec<G::ScalarField>,
+}
```

Comment:
so a few comment here:

* can you move the data structure to the top of the file?
* can you document on each fields? since they are public. (e.g. "the [VerifierIndex] associated to this proof", "The proof to verify", "the public input used in the creation of this proof")
* I don't think `BatchedProof` is a good name, usually a batch is several proofs. I'm not sure what's a better name though... `ProofPackage`? `BundleToVerify`? `ReadyToVerify`? ...

Reply:
Funny thing: there's already Context struct, that only lacks public input. I think I can refactor this a little more. Stay tuned for the follow-up PR.

Reply:
See #982